haystack/preview/utils/filters.py

```diff
@@ -33,7 +33,7 @@ def and_operation(conditions: List[Any], document: Document, _current_key: str):
     :return: True if the document matches all the filters, False otherwise
     """
     for condition in conditions:
```

Comment:
That's a good example of where `all` can be used by the way: `return all(document_matches_filter(conditions=condition, document=document, _current_key=_current_key) for condition in conditions)`